internal/keystore/kes/kes.go

```
@@ -50,17 +55,34 @@ func Connect(ctx context.Context, config *Config) (*Conn, error) {
 	if len(config.Endpoints) == 0 {
 		return nil, errors.New("kes: no endpoints provided")
 	}
-	if config.Certificate == "" {
-		return nil, errors.New("kes: no certificate provided")
-	}
-	if config.PrivateKey == "" {
-		return nil, errors.New("kes: no private key provided")
+	if config.APIKey != nil && (config.PrivateKey != "" || config.Certificate != "") {
+		return nil, errors.New("kes: ambiguous configuration: API key as well as mTLS private key and certificate provided")
 	}
 
-	cert, err := https.CertificateFromFile(config.Certificate, config.PrivateKey, "")
-	if err != nil {
-		return nil, err
+	var (
+		err  error
+		cert tls.Certificate
+	)
+	switch {
+	case config.APIKey != nil:
+		cert, err = kes.GenerateCertificate(config.APIKey)
+		if err != nil {
+		}
```

can we return the error here?